### PR TITLE
feat: now survives messages and not just segments being cr/lf terminated

### PR DESCRIPTION
**Pull request by plisitza**
to tolerate the fact that the pharmacy team sometimes outputs this way.

### CODE REVIEW

File: commons/files.go [hunk 1]
```diff
@@ -8,7 +8,8 @@ import (
 	"regexp"
 )
 
-var hl7SplitToken = regexp.MustCompile("(\\r(\\n|\\x1c)+(\\n\\r)?|$)")
+var hl7SplitToken = regexp.MustCompile("(\\r(\\n|\\x1c)+(\\n\\r)?MSH\\||$)")
+var hl7FindStartToken = regexp.MustCompile("(MSH\\||$)")
```
Comment (ghost) [inline commons/files.go]: Suggestion no change to hl7SplitToken.
Change hl7FindStartToken to:
````var hl7SplitTokenVFP = regexp.MustCompile("ZXO\\r(\\n)?|$")````

File: commons/files.go [hunk 2]
```diff
@@ -8,7 +8,8 @@ import (
...
 
 const scanBufferSize = 10 * 1024 * 1024
```
Comment (ghost) [inline commons/files.go:15]: Increase buffer to 64 * 1024 * 1024?

File: commons/files.go [hunk 3]
```diff
@@ -38,7 +39,8 @@ func crLfSplit(data []byte, atEOF bool) (advance int, token []byte, err error) {
 	} else {
 		loc := hl7SplitToken.FindIndex(data) // found record delimiter
```
Comment (ghost) [inline commons/files.go:41]: Suggestion to change `else` to:
````         
                loc := hl7SplitTokenVFP.FindIndex(data) // found VFP record delimiter
                if loc == nil {
                        loc = hl7SplitToken.FindIndex(data) // found record delimiter
                }
                if loc != nil || atEOF {
                        hl7RecPatch := []byte(strings.ReplaceAll(string(data[0:loc[0]]),"\r\n","\r")) // cr/lf found after each segment, patch.
                        return loc[1], hl7RecPatch, nil
                }
````

File: commons/files.go (hunk 1 above)
Comment (plisitza) [reply commons/files.go]: yeah that won't work, as discussed.
elaborating further on MSH might, though? e.g. ... "MSH\\|^~\\\&\\||$"?

File: commons/files.go (hunk 2 above)
Comment (plisitza) [reply commons/files.go:15]: no that's not the issue. there is no record that exceeds 10mb. this just punts to a longer length. the real issue is that this does not like records that do not end in the buffer? which will happen for any file longer than the buffer (like most epic files are, even compared to 64mb?)

Comment (plisitza) [reply commons/files.go:15]: you can test this yourself by increasing the buffer to what you suggest and then processing a file bigger than that?

File: commons/files.go (hunk 3 above)
Comment (plisitza) [reply commons/files.go:41]: good idea, done.

**Review by ghost — APPROVED** [review]

**Review by g-yhuang — APPROVED** [review]

**Pull request merged**